File: src/test/java/LongestCommonPrefixTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class LongestCommonPrefixTest {
    @Test
    void should_find_longest_prefix() {
        LongestCommonPrefix solution = new LongestCommonPrefix();
        String prefix = solution.longestCommonPrefix(new String[]{"flower", "flow", "flight"});
        assertEquals("fl", prefix);
    }

    @Test
    void should_find_longest_prefix2() {
        LongestCommonPrefix solution = new LongestCommonPrefix();
        String prefix = solution.longestCommonPrefix(new String[]{"flower", "flow", "flowerist"});
        assertEquals("flow", prefix);
    }

    @Test
    void should_return_empty_for_no_prefix() {
        LongestCommonPrefix solution = new LongestCommonPrefix();
        String prefix = solution.longestCommonPrefix(new String[]{"dog","racecar","car"});
        assertEquals("", prefix);
    }

    @Test
    void should_return_empty_for_not_all_prefix() {
        LongestCommonPrefix solution = new LongestCommonPrefix();
        String prefix = solution.longestCommonPrefix(new String[]{"dog","racecar","dar"});
        assertEquals("", prefix);
    }

    @Test
    void should_return_for_same_values() {
        LongestCommonPrefix solution = new LongestCommonPrefix();
        String prefix = solution.longestCommonPrefix(new String[]{"dog","dog"});
        assertEquals("dog", prefix);
    }

    @Test
    void should_return_single_value() {
        LongestCommonPrefix solution = new LongestCommonPrefix();
        String prefix = solution.longestCommonPrefix(new String[]{"dog"});
        assertEquals("dog", prefix);
    }
}

File: src/test/kotlin/LongestCommonPrefixTest.kt
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class LongestCommonPrefixTest {
    @Test
    fun should_find_longest_prefix() {
        val solution = LongestCommonPrefix()
        val prefix = solution.longestCommonPrefix(arrayOf("flower", "flow", "flight"))
        Assertions.assertEquals("fl", prefix)
    }

    @Test
    fun should_find_longest_prefix2() {
        val solution = LongestCommonPrefix()
        val prefix = solution.longestCommonPrefix(arrayOf("flower", "flow", "flowerist"))
        Assertions.assertEquals("flow", prefix)
    }

    @Test
    fun should_return_empty_for_no_prefix() {
        val solution = LongestCommonPrefix()
        val prefix = solution.longestCommonPrefix(arrayOf("dog", "racecar", "car"))
        Assertions.assertEquals("", prefix)
    }

    @Test
    fun should_return_empty_for_not_all_prefix() {
        val solution = LongestCommonPrefix()
        val prefix = solution.longestCommonPrefix(arrayOf("dog", "racecar", "dar"))
        Assertions.assertEquals("", prefix)
    }

    @Test
    fun should_return_for_same_values() {
        val solution = LongestCommonPrefix()
        val prefix = solution.longestCommonPrefix(arrayOf("dog", "dog"))
        Assertions.assertEquals("dog", prefix)
    }

    @Test
    fun should_return_single_value() {
        val solution = LongestCommonPrefix()
        val prefix = solution.longestCommonPrefix(arrayOf("dog"))
        Assertions.assertEquals("dog", prefix)
    }
}
